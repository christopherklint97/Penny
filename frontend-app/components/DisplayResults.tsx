interface DisplayResultsProps {
  cities: CityProps[]
  handleClick: (event: any) => Promise<void>
  select: any
}

export interface CityProps {
  place_id?: string
  description?: string
}

export default function DisplayResults({
  cities,
  handleClick,
  select,
}: DisplayResultsProps) {
  const cityCards = cities.map((city) => {
    if (select.place_id === city.place_id) {
      return (
        <div
          className="justify-center mb-2 text-left align-middle bg-yellow-100 rounded-md shadow-md cursor-pointer w-60"
          key={city.place_id}
        >
          <p
            className="p-2 mx-auto text-sm"
            id={city.place_id}
            onClick={handleClick}
          >
            {city.description}
          </p>
        </div>
      )
    } else {
      return (
        <div
          className="justify-center mb-2 text-left align-middle rounded-md shadow-md cursor-pointer hover:bg-gray-200 w-60 bg-gray-50"
          key={city.place_id}
        >
          <p
            className="p-2 mx-auto text-sm"
            id={city.place_id}
            onClick={handleClick}
          >
            {city.description}
          </p>
        </div>
      )
    }
  })

  return <div>{cityCards}</div>
}
