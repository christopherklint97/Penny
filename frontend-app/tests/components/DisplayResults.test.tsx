import React from 'react'
import { render } from '../testUtils'
import DisplayResults from '../../components/DisplayResults'

describe('DisplayResults component', () => {
  it('renders without crashing', function () {
    render(
      <DisplayResults handleClick={jest.fn()} cities={[]} select={jest.fn()} />
    )
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <DisplayResults handleClick={jest.fn()} cities={[]} select={jest.fn()} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
