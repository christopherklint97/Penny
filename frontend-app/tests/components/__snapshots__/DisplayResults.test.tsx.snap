// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayResults component matches snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
